go/stats/prombackend/prombackend.go

```diff
@@ -0,0 +1,63 @@
+package prombackend
+
+import (
+	"net/http"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"vitess.io/vitess/go/stats"
+)
+
+// PromBackend implements PullBackend using Prometheus as the backing metrics storage
+type PromBackend struct {
+	namespace string
+}
+
+func InitPromBackend(namespace string) {
+	http.Handle("/metrics", promhttp.Handler())
+	promBackend := NewPromBackend(namespace)
+	stats.RegisterPullBackendImpl("prom", promBackend)
+}
+
+func NewPromBackend(namespace string) *PromBackend {
+	return &PromBackend{namespace: namespace}
+}
+
+// NewCounter is part of the PullBackend interface
+func (be *PromBackend) NewMultiCounter(mc *stats.MultiCounters, name string) {
+	// TODO: camelcase => snakecase here for name
+	c := NewMultiCountersCollector(map[string]*prom.Desc{
```

Comment:
It seems to me that if you stash the pointer to `mc` as part of the multiCountersCollector, then in the `Collect` implementation you don't need to go back again and call `stats.GetMultiCounter(name)` since you can just have the pointer to the underlying counter in the multiCountersCollector itself.